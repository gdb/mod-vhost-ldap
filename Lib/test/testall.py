# testall.py -- a regression test for the Python interpreter.
# To run the tests, execute "import testall" in a clean interpreter.
# It is a good idea to do this whenever you build a new interpreter.
# Remember to add new tests when new features are added!

import sys
from test_support import *

print 'test_grammar'
forget('test_grammar')
import test_grammar

for t in ['test_opcodes', 'test_operations', 'test_builtin',
	  'test_exceptions', 'test_types', 'test_math', 'test_time',
	  'test_array', 'test_strop', 'test_md5', 'test_cmath',
	  'test_crypt', 'test_dbm', 'test_new',
	  ]:
    print t
    unload(t)
    try:
	__import__(t, globals(), locals())
    except ImportError, msg:
	sys.stderr.write('%s.  Uninstalled optional module?\n' % msg)

print 'Passed all tests.'
