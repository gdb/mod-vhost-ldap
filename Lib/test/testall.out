test_grammar
1. Parser
1.1 Tokens
1.1.1 Backslashes
1.1.2 Numeric literals
1.1.2.1 Plain integers
1.1.2.2 Long integers
1.1.2.3 Floating point
1.1.3 String literals
1.2 Grammar
single_input
file_input
expr_input
eval_input
funcdef
simple_stmt
expr_stmt
print_stmt
1 2 3
1 2 3
1 1 1
del_stmt
pass_stmt
flow_stmt
break_stmt
continue_stmt
return_stmt
raise_stmt
import_stmt
global_stmt
exec_stmt
if_stmt
while_stmt
for_stmt
try_stmt
suite
test
comparison
binary mask ops
shift ops
additive ops
multiplicative ops
unary ops
selectors
atoms
classdef
test_opcodes
2. Opcodes
XXX Not yet fully implemented
2.1 try inside for loop
2.2 raise class exceptions
test_operations
3. Operations
XXX Not yet implemented
test_builtin
4. Built-in functions
test_b1
__import__
abs
apply
callable
chr
cmp
coerce
compile
delattr
dir
divmod
eval
execfile
filter
float
getattr
hasattr
hash
hex
id
int
len
long
map
max
min
test_b2
oct
open
ord
pow
range
input and raw_input
testing
testing
reduce
reload
repr
round
setattr
str
tuple
type
vars
xrange
test_exceptions
5. Built-in exceptions
AttributeError
EOFError
IOError
ImportError
IndexError
KeyError
KeyboardInterrupt
(not testable in a script)
MemoryError
(not safe to test)
NameError
OverflowError
RuntimeError
(not used any more?)
SyntaxError
SystemError
(hard to reproduce)
SystemExit
TypeError
ValueError
ZeroDivisionError
test_types
6. Built-in types
6.1 Truth value testing
6.2 Boolean operations
6.3 Comparisons
6.4 Numeric types (mostly conversions)
6.4.1 32-bit integers
6.4.2 Long integers
6.4.3 Floating point numbers
6.5 Sequence types
6.5.1 Strings
6.5.2 Tuples
6.5.3 Lists
6.5.3a Additional list operations
6.6 Mappings == Dictionaries
test_math
math module, testing with eps 1e-05
constants
acos
asin
atan
atan2
ceil
cos
cosh
exp
fabs
floor
fmod
frexp
hypot
ldexp
log
log10
modf
pow
sin
sinh
sqrt
tan
tanh
test_time
test_array
test_strop
test_md5
MD5 test suite:
MD5 ("") = d41d8cd98f00b204e9800998ecf8427e
MD5 ("a") = 0cc175b9c0f1b6a831c399e269772661
MD5 ("abc") = 900150983cd24fb0d6963f7d28e17f72
MD5 ("message digest") = f96b697d7cb7938d525a2f31aaf161d0
MD5 ("abcdefghijklmnopqrstuvwxyz") = c3fcd3d76192e4007dfb496cca67e13b
MD5 ("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789") = d174ab98d277d9f5a5611c2c9f419d9f
MD5 ("12345678901234567890123456789012345678901234567890123456789012345678901234567890") = 57edf4a22be3c955ac49da2e2107b67a
test_cmath
test_crypt
Test encryption:  abeTm2pJKypwA
test_new
Passed all tests.
